### ISSUE TITLE
Freeze in MGLOfflineStorage when initializing MGLMapView: iOS

### ISSUE THREAD

**Issue by akatkov7** [mapbox/mapbox-gl-native#11914]
**Platform:** iOS 10.3
**Mapbox SDK version:** 4.0.1 (but also had this issue on 3.7)

### Steps to trigger behavior

Not sure if the exact trigger, but the issue arises during normal development where I am redeploying to my dev phone and eventually gets frozen with below stack trace when an MGLMapView is created.

### Expected behavior

Creating an MGLMapView should not freeze the thread it is on and initialize properly.

### Actual behavior

Freezes the thread it is on and never completes.

Here is a screenshot of the stack trace when the freeze occurs:
<img width="909" alt="image" src="https://user-images.githubusercontent.com/6496359/40077318-447cd348-584f-11e8-9415-759b817accc4.png">

Initialization occurs inside of viewDidLoad like this:
```swift
let mapView = MGLMapView(frame: UIScreen.main.bounds)
mapView.styleURL = URL(string: "my style url")
mapView.setCenter(CLLocationCoordinate2D(latitude: 59.31, longitude: 18.06), zoomLevel: 9, animated: false)
mapView.isHidden = true
```

Reinstalling the app doesn't help, only a phone reboot fixes it. I have no prior experience with Mapbox (but do with Google Maps integration) so maybe there is something I'm not doing correctly. A brief look into the source looks like the issue is occurring when the DefaultFileSource is created inside MGLOfflineStorage.init (https://github.com/mapbox/mapbox-gl-native/blob/62c875e01b07197024e3806e8b2882160ce1195c/platform/darwin/src/MGLOfflineStorage.mm#L214). However, my C++ isn't great so not sure exactly where the issue might be happening beyond that during the file source creation. The cache.db file did exist and I can provide that if needed.

I'm not using MGLOfflineStorage in my code at all either.

If there is any other info that would be useful to provide, let me know. Thanks for any help you can provide.

**Comment by friedbunny** [comment]
Thanks for the report, @akatkov7. It looks like you have a catch-all exception breakpoint enabled, which is stopping at the initialing exception for the database — #10796. This flow has been refactored for `ios-v4.1.x`, but, in the meantime, please hit continue when the breakpoint is triggered or use more specific breakpoint.

**Comment by akatkov7** [comment]
@friedbunny Thanks for the feedback. However, the stack trace you see comes when I manually pause program execution in Xcode, not an automatic breakpoint that is being triggered. Is there a beta build available of 4.1.x yet to see if it would help with this issue?

**Comment by friedbunny** [comment]
@akatkov7 I see, then let’s reopen. There are no builds of `ios-v4.1.0` published yet, so please stay tuned for that. If you could post the full text of this stack trace with all of the threads, that may help narrow this down.

**Comment by jfirebaugh** [comment]
This may have been fixed by #12151. If you still experience this with v4.2.0, please provide the full text of this stack trace with all of the threads.
